File: app/src/main/java/in/ceeq/define/DefineApplication.java
package in.ceeq.define;

import android.app.Application;

import in.ceeq.define.utils.AnalyticsUtils;

public class DefineApplication extends Application {

    @Override
    public void onCreate() {
        super.onCreate();
        new AnalyticsUtils().init(this);
    }
}

File: app/src/main/java/in/ceeq/define/DefineApplication.kt
package `in`.ceeq.define

import `in`.ceeq.define.di.component.DaggerAppComponent
import `in`.ceeq.define.utils.AnalyticsUtils
import android.app.Application
import dagger.android.AndroidInjector
import dagger.android.DispatchingAndroidInjector
import dagger.android.HasAndroidInjector
import javax.inject.Inject


class DefineApplication : Application(), HasAndroidInjector {

    @Inject
    lateinit var androidInjector: DispatchingAndroidInjector<Any>

    override fun onCreate() {
        super.onCreate()
        AnalyticsUtils().init(this)

        DaggerAppComponent.builder()
                .app(this)
                .build().inject(this)
    }

    override fun androidInjector(): AndroidInjector<Any> = androidInjector
}
